phoenix-core-client/src/main/java/org/apache/phoenix/compile/FromCompiler.java

```diff
@@ -280,7 +281,8 @@ public static ColumnResolver getResolver(NamedTableNode tableNode, PhoenixConnec
 
     public static ColumnResolver getResolver(SingleTableStatement statement, PhoenixConnection connection)
             throws SQLException {
-        SingleTableColumnResolver visitor = new SingleTableColumnResolver(connection, statement.getTable(), true);
+        SingleTableColumnResolver visitor
+                = new SingleTableColumnResolver(connection, statement.getTable(), true, true);
```

Comment:
Why are we setting alwaysHitServer to true ?

Reply:
@tkhurana This resolver is also used in several DDL paths. 
We create resolvers for query/mutations in `getResolverForQuery` and `getResolverForMutation` and that is where we set it to false.